sympy/matrices/expressions/matmul.py

```diff
@@ -10,9 +10,10 @@
 from sympy.matrices.expressions.matexpr import (MatrixExpr, ShapeError,
         Identity, ZeroMatrix)
 from sympy.matrices.matrices import MatrixBase
+from sympy.core.operations import AssocOp
 
 
-class MatMul(MatrixExpr):
+class MatMul(MatrixExpr, AssocOp):
```

Comment:
I ask because AssocOp subclasses must define `Identity`. Sadly, the way the class is defined, it's a class attribute (and even if it were an instance attribute, a MatMul has two identities, left and right). I think unless we make some modifications, it isn't possible to use AssocOp with MatMul.